server.js

```diff
@@ -711,6 +711,19 @@ const envVarsCheck = callback => {
   }
 };
 
+const couchDbNoAdminPartyModeCheck = callback => {
+  const noAuthUrl = process.env.COUCH_URL.replace(/\/\/[^@]+@/, '//'),
+        protocol = noAuthUrl.split('://')[0];
```

Comment:
It'd be great if we could start to move away from regexing URLs as it's hard to read and potentially error prone. How about using the URL api? Something like this...

```js
const url = require('url');
const noAuthCouchUrl = new url(process.env.COUCH_URL);
noAuthCouchUrl.username = null;
noAuthCouchUrl.password = null;
const net = require(noAuthCouchUrl.protocol);
net.get(noAuthUrl.toString(), err => {

```